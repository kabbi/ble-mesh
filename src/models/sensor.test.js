const { typeSet } = require("./sensor");

const { hex } = require("../utils/tags");
const { parse } = require("../utils/binary")({
  'jBinary.littleEndian': true,
  ...typeSet
});

describe("models/sensor", () => {
  it("parses sensor status message with two values", () => {
    const data = parse("SensorStatus", hex`820a540ba20e6810`);
    expect(data).toMatchSnapshot();
  });
});
