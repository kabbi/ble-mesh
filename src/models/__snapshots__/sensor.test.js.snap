// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/sensor parses sensor status message with two values 1`] = `
Array [
  Object {
    "propertyId": "PresentDeviceOperatingTemperature",
    "value": 29,
  },
  Object {
    "propertyId": "PrecisePresentAmbientTemperature",
    "value": 42,
  },
]
`;
